### ISSUE TITLE
About SliceGAN

### ISSUE THREAD

**Issue by asadullah1120** [stke9/SliceGAN#12]
Hi
I'm glad to meet you.

My name is Asadullah.
I'm master student in Tokyo.

I ran with your code that you provided and generation with the images  you use worked fine.
However, I tried to generate to 3D structure myself using the images 3 pieces and could not get it to work.
If you know how to do it, could you please let me know.

**Comment by stke9** [comment]
Asadullah,

Send me your run_slicegan.py file settings and your 3 images. When you say it didn’t work what does this mean? If error messages, send those too.

Best,
Steve

From: asadullah1120 ***@***.***>
Date: Friday, 20 May 2022 at 11:44
To: stke9/SliceGAN ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [stke9/SliceGAN] About SliceGAN (Issue #12)
This email from ***@***.*** originates from outside Imperial. Do not click on links and attachments unless you recognise the sender. If you trust the sender, add them to your safe senders list<https://spam.ic.ac.uk/SpamConsole/Senders.aspx> to disable email stamping for this address.



Hi
I'm glad to meet you.

My name is Asadullah.
I'm master student in Tokyo.

I ran with your code that you provided and generation with the images you use worked fine.
However, I tried to generate to 3D structure myself using the images 3 pieces and could not get it to work.
If you know how to do it, could you please let me know.

**Comment by asadullah1120** [comment]
hi

sorry for the late reply.

I made train.py and  test.py based on your run_slicegan.py.
Below  is your file.



### Welcome to SliceGAN ###
####### Steve Kench #######
'''
Use this file to define your settings for a training run, or
to generate a synthetic image using a trained generator.
'''

from slicegan import model, networks, util
import argparse
# Define project name
Project_name = 'NMC_exemplar_final'
# Specify project folder.
Project_dir = 'Trained_Generators/NMC'
# Run with False to show an image during or after training
parser = argparse.ArgumentParser()
parser.add_argument('training', type=int)
args = parser.parse_args()
Training = args.training
Project_path = util.mkdr(Project_name, Project_dir, Training)

## Data Processing
# Define image  type (colour, grayscale, three-phase or two-phase.
# n-phase materials must be segmented)
image_type = 'threephase'
# define data type (for colour/grayscale images, must be 'colour' / '
# greyscale. nphase can be, 'tif', 'png', 'jpg','array')
data_type = 'tif'
# Path to your data. One string for isotrpic, 3 for anisotropic
data_path = ['Examples/NMC.tif']

## Network Architectures
# Training image size, no. channels and scale factor vs raw data
img_size, img_channels, scale_factor = 64, 3,  1
# z vector depth
z_channels = 16
# Layers in G and D
lays = 6
# kernals for each layer
dk, gk = [4]*lays, [4]*lays
# strides
ds, gs = [2]*lays, [2]*lays
# no. filters
df, gf = [img_channels,64,128,256,512,1], [z_channels,512,256,128,64,img_channels]
# paddings
dp, gp = [1,1,1,1,0],[2,2,2,2,3]

## Create Networks
netD, netG = networks.slicegan_nets(Project_path, Training, image_type, dk, ds, df,dp, gk ,gs, gf, gp)

# Train
if Training:
    model.train(Project_path, image_type, data_type, data_path, netD, netG, img_channels, img_size, z_channels, scale_factor)
else:
    img, raw, netG = util.test_img(Project_path, image_type, netG(), z_channels, lf=6, periodic=[0, 1, 1])

**Comment by stke9** [comment]
What is not working please?

**Comment by asadullah1120** [comment]
I use this three images.
I used these three images to train them, but they over-learned.


![10](https://user-images.githubusercontent.com/105913058/169806163-8e4e3551-5691-4160-9b03-e38aaadd6fa6.png)
![11](https://user-images.githubusercontent.com/105913058/169806170-3ec6899f-232d-4635-b147-0f77985ac745.png)
![12](https://user-images.githubusercontent.com/105913058/169806172-5ec0b1ea-18ee-4bbb-b803-cdf456182c34.png)
![414_GpGraph](https://user-images.githubusercontent.com/105913058/169806175-3d50cfbc-4993-4b03-b4c2-c0124cebaaae.png)
![414_LossGraph](https://user-images.githubusercontent.com/105913058/169806179-4797b829-ec67-4a76-9ad4-f1864a9d12e9.png)
![414_slices](https://user-images.githubusercontent.com/105913058/169806185-ca155901-c585-48c4-a442-3450e76571d7.png)
![414_WassGraph](https://user-images.githubusercontent.com/105913058/169806190-f20d8c2a-2def-4f9c-a294-328e1d074a30.png)

**Comment by stke9** [comment]
As you mention, this definitely looks like overfitting. You can try stop training after 10k iterations, which looks like the point where it really gets bad. You should then generate a large cube and look at the quality in ImageJ. Unfortunately you really have very little data here, but it might work OK. If not, you need bigger images for 2 and 3 (1 is definitely fine). Just to check, is your microstructure anisotropic in all 3 directions?

**Comment by asadullah1120** [comment]
I want to change the size of the cube, where do I change the program.

**Comment by stke9** [comment]
It depends what you mean - do you want to change the size of the cube during training (this can be very computationally expensive and is a bit complex) or after training (this is very easy, on line 54 just change lf, from 6 to a higher number and run with python run_slicegan.py 0)

**Comment by asadullah1120** [comment]
Thank you for answering my question.
I have one more question.
I have created train.py and test.py based on your run_slicegan.py.
My train.py learn fine, but I created test.py and programmed it to be able to generate 3D images.
However, I get an error that the file exists but doesn't exist.
![Screenshot from 2022-05-31 15-49-38](https://user-images.githubusercontent.com/105913058/171110629-153d0828-04ac-4cf3-92be-dbae65263045.png)
![Screenshot from 2022-05-31 15-49-47](https://user-images.githubusercontent.com/105913058/171110656-5942a81a-8283-439a-92ee-86b16f1790f8.png)
![Screenshot from 2022-05-31 15-00-01](https://user-images.githubusercontent.com/105913058/171110681-9b29467d-4e23-4af2-bce2-9555922ddd46.png)
![Screenshot from 2022-05-31 15-08-11](https://user-images.githubusercontent.com/105913058/171110695-009b7796-349f-4759-b80a-c7916a276294.png)
